app/unit-tests/src/org/commcare/android/util/TestUtils.java

```diff
@@ -103,15 +104,15 @@ public TransactionParser getParser(KXmlParser parser) {
                     //there's no good lifecycle to manage the bulk processor in, but at least
                     //this will validate that the bulk processor works.
                     if(bulkProcessingEnabled)  {
-                        return new AndroidBulkCaseXmlParser(parser, getCaseStorage(db), new EntityStorageCache("case", db), new AndroidCaseIndexTable(db)) {
+                        return new AndroidBulkCaseXmlParser(parser, getCaseStorage(db), new EntityStorageCache("case", db, ReportingUtils.getAppId()), new AndroidCaseIndexTable(db)) {
```

Comment:
Same comment as above WRT reportingUtils. Very important case-in-point: ReportingUtils fails by producing a blank string. We want to fail with a real exception here that bubbles up, since these methods won't work without a real id.

Reply:
@ctsims After making this change, `processResourceTransaction(String resourcePath)` is failing due to `java.lang.RuntimeException: No App installed!!!` exception while getting appID at a bunch of places. Is this expected and we need to fail silently here or we need to change the tests to install an app first before calling `processResourceTransaction`.

Reply:
Hm, my guess is that the fundamental data type (EntityStoreCache) should be allowed to get created with no App ID, and if it's going to fail that it should fail laziy, or that we should not load the entity cache if the app ID isn't available.

there are operations which are possible without the app ID, it might be the case that we should allow it to be loaded in a way where those functions work, but anything that relies on an appID fails if you try to use them in that mode.

Reply:
I am now passing entityCache as a Nullable and checking to see if it's exist to get past these errors. Though not sure if that's the right way to go based on your above comment.

Reply:
I think this is essentially equivalent to what we were doing previously, so this looks good to me for now. 

I'm a bit confused though, I thought that the appID being null was what was causing those issues. Was it actually that the entityCache itself the source of problems?

Reply:
oh, whoops, misread that. Re-reviewing now, I think there's one change to make.